lib/useKResponsiveWindow/index.js

```diff
@@ -1,132 +1,134 @@
-    heightQuery.startListening();
-    for (const widthQuery of widthQueries) {
-      widthQuery.startListening();
-    }
-  };
+/**
+ * Stop listening to media query changes
+ */
+function stopListening() {
+  orientationQuery.stopListening();
+  heightQuery.stopListening();
+};
 
-  /**
-   * Stop listening to media query changes
-   */
-  const stopListening = () => {
-    orientationQuery.stopListening();
-    heightQuery.stopListening();
-    for (const widthQuery of widthQueries) {
-      widthQuery.stopListening();
-    }
-  };
 
-  /**
-   * Update window gutter
-   */
-  const updateGutter = () => {
-    if (windowIsSmall.value) {
-      windowGutter.value = 16;
-    } else if (smallestWindowDimensionIsLessThan(600)) {
-      windowGutter.value = 16;
-    } else {
-      windowGutter.value = 24;
-    }
-  };
+function updateBreakPoint() {
+  const SCROLL_BAR = 16;
+  const widthBreakpoints = [
+    480, 600, 840, 960 - SCROLL_BAR, 1280 - SCROLL_BAR, 1440 - SCROLL_BAR, 1600 - SCROLL_BAR,
+  ];
 
-  /**
-   * Check if the smallest window dimension(width or height) is smaller than the specified dimension
-   * @param {Number} dimension in px
-   * @returns {Boolean}
-   */
-  const smallestWindowDimensionIsLessThan = dimension => {
-    return (
-      windowBreakpoint.value < 4 && Math.min(windowWidth.value, windowHeight.value) < dimension
-    );
-  };
+  // if windowWidth is null (upon initialization), breakpoint = 0
+  const firstMatchingWidthIndex = windowWidth.value
+    ? widthBreakpoints.findIndex(matchingWidth => windowWidth.value < matchingWidth)
+    : 0;
+
+  // if windowWidth > 1585, breakpoint = 7
+  const matchingBreakpoint = firstMatchingWidthIndex >= 0
+    ? firstMatchingWidthIndex
+    : 7;
+
+  windowBreakpoint.value = matchingBreakpoint;
+  updateGutter();
+};
+
+watch([windowWidth, windowHeight], updateBreakPoint);
 
-  /**
-   * Update window orientation
-   * @param {Boolean} portrait
-   */
-  const updateOrientation = portrait => {
-    windowIsPortrait.value = portrait;
-    windowIsLandscape.value = !windowIsPortrait.value;
-  };
 
+function updateGutter() {
+  if (windowIsSmall.value || smallestWindowDimensionIsLessThan(600)) {
+    windowGutter.value = 16;
+  } else {
+    windowGutter.value = 24;
+  }
+};
+
+/**
+ * Check if the smallest window dimension(width or height) is smaller than the specified dimension
+ * @param {Number} dimension in px
+ * @returns {Boolean}
+ */
+function smallestWindowDimensionIsLessThan(dimension) {
+  return (
+    windowBreakpoint.value < 4 && Math.min(windowWidth.value, windowHeight.value) < dimension
+  );
+};
+
+/**
+ * Update window orientation
+ * @param {Boolean} portrait
+ */
+function updateOrientation(portrait) {
+  windowIsPortrait.value = portrait;
+  windowIsLandscape.value = !windowIsPortrait.value;
+};
+
+/**
+ * Export window properties
+ * @returns {Object} Object with window properties
+ */
+export default function useKResponsiveWindow() {
```

Comment:
So, the other thing that is happening here, is that the onMounted and onBeforeUnMount hooks for useKWindowDimension are never being called, because we've moved the invocation into the module scope.

So, to make sure they do get initialized, we can just make a dummy call to these in the function scope as well:

```
// Do this to ensure that we call the useKWindowDimension lifecycle hooks
useKWindowDimension();
```